src/core/handlers/GraphQLHandler.ts

```diff
@@ -31,6 +31,23 @@ export interface GraphQLHandlerInfo extends RequestHandlerDefaultInfo {
   operationName: GraphQLHandlerNameSelector
 }
 
+export type GraphQLRequestParsedResult = {
+  match: Match
+} & (
+  | ParsedGraphQLRequest<GraphQLVariables>
+  /**
+   * An empty version of the ParsedGraphQLRequest
+   * which simplifies the return type of the resolver
+   * when the request is to a non-matching endpoint
+   */
+  | {
```

Comment:
we could also nest instead of merging match and these properties. I don't have a strong preference, besides not wanting to name the object containing these here.

it could be `graphqlAttributes` or `parsedGraphqlRequest`  but this is already a parsed result, so it feels a bit weird

Reply:
As long as the existing typings tests pass, I have no objection how we handle this.